src/routes/movement-detection/MovementDetection.js

```diff
@@ -0,0 +1,149 @@
+import React, { useState, useCallback } from 'react';
+import { useSelector, shallowEqual, useDispatch } from 'react-redux';
+import withStyles from 'isomorphic-style-loader/withStyles';
+import map from 'lodash/map';
+import filter from 'lodash/filter';
+import reduce from 'lodash/reduce';
+import property from 'lodash/property';
+import Aside from './Aside/Aside';
+import Info from './Info/Info';
+import Sensor from './Sensor/Sensor';
+import addImgUrl from '../../actions/movementDetections/img';
+import {
+  updateSensors,
+  disarmSensors,
+} from '../../actions/movementDetections/sensors';
+import s from './MovementDetection.css';
+
+function MovementDetection() {
+  const dispatch = useDispatch();
+  const { imgUrl, sensors, isDisarmed } = useSelector(
+    property('movementDetection'),
+    shallowEqual,
+  );
+  const [isEdit, setIsEdit] = useState(false);
+  const [localState, setLocalState] = useState({ imgUrl, sensors });
+
+  const { imgUrl: imgToRender, sensors: sensorsToRender } = isEdit
+    ? localState
+    : { imgUrl, sensors };
+
+  const positionHandler = useCallback(
+    (ui, index) => {
+      const nextState = {
+        ...localState,
+        sensors: Array.from(localState.sensors),
+      };
+      nextState.sensors[index] = {
+        ...nextState.sensors[index],
+        x: ui.x,
+        y: ui.y,
+      };
+      setLocalState(nextState);
+    },
+    [localState],
+  );
+
+  const fileHandler = useCallback(
+    event => {
+      const fileReader = new FileReader();
+      fileReader.readAsDataURL(event.target.files[0]);
+      fileReader.addEventListener('load', () => {
+        setLocalState({ ...localState, imgUrl: fileReader.result });
+      });
+    },
+    [localState],
+  );
+
+  function disarmHandler() {
+    dispatch(disarmSensors());
+  }
+  function editHandler() {
+    setIsEdit(!isEdit);
+    setLocalState({ imgUrl, sensors: Array.from(sensors) });
+  }
+  function saveHandler() {
+    setIsEdit(false);
+    dispatch(updateSensors(localState.sensors));
+    dispatch(addImgUrl(localState.imgUrl));
+  }
+  function cancelHandler() {
+    setIsEdit(false);
+    setLocalState({ imgUrl, sensors });
+  }
+
+  function addSensorHandler() {
+    const nextSensorId =
+      reduce(
```

Comment:
I'm happy that you utilize reduce's full power, but consider rewriting it with `lodash/maxBy` to gain more readability